xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/store/TemporaryAttachmentManager.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.store;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.servlet.http.Part;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.doc.XWikiAttachment;
+
+/**
+ * Interface for operations related to temporary upload of attachments.
+ * The idea of this API is to allow obtaining directly a temporary {@link XWikiAttachment} from a {@link Part} and to
+ * keep it in cache until it's saved. The cache is managed per user session, and per edited document.
+ *
+ * @version $Id$
+ * @since 14.3RC1
+ */
+@Unstable
+@Role
+public interface TemporaryAttachmentManager
```

Comment:
The interface name should maybe make more clear that it's related to the session, to not be missleading.

Reply:
I put a note in the javadoc for it, but I could add a bit more explanations probably. 
Now for renaming, do you have ideas? Something like `TemporaryAttachmentSessionManager`?

Reply:
Yes I saw the javadoc, but I think it's not enough. `TemporaryAttachmentSessionManager` sounds good.